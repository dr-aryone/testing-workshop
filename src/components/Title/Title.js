import React from "react";
import PropTypes from "prop-types";

const Title = ({ title }) => <h2 data-testid="titleContainer">{title}</h2>;

Title.propTypes = {
  title: PropTypes.string
};

export default Title;
