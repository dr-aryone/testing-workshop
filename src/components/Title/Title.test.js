import React from "react";
import ReactDOM from "react-dom";
import { render } from "@testing-library/react";
import Title from "./Title";

describe("Title Page", () => {
  it("Renders without crashing", () => {
    ReactDOM.render(<Title title="TitleTest" />, document.createElement("div"));

    ReactDOM.unmountComponentAtNode(document.createElement("div"));
  });

  it("Loads component", () => {
    const { getByTestId } = render(<Title title="TitleTest" />);

    const title = getByTestId("titleContainer");

    expect(title).toBeInTheDocument();
  });

  it("Match Snapshot", () => {
    const { firstChild } = render(<Title title="TitleTest" />);

    expect(firstChild).toMatchSnapshot();
  });

  it("Renders the title", () => {
    const { getByText } = render(<Title title="TitleTest" />);

    const title = getByText("TitleTest");

    expect(title).toBeInTheDocument();
  });
});
