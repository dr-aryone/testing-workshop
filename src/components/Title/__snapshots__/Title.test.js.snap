// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title Page Match Snapshot 1`] = `undefined`;
